Update component “dropdown-selects” (#71)
Zywave/booster: src/_areas/design-system/components/dropdown-selects.md

---
title: Dropdown selects
subtitle: Type of input field in which a user fills an input box with an item
  from a list of acceptable items.
api: https://cdn.zywave.com/@zywave/zui-select@next/dist/custom-elements.json
demo: https://cdn.zywave.com/@zywave/zui-select@next/demo/index.html
mainComponentName: zui-select-dropdown
includedElements:
  - zui-select-dropdown
  - zui-option
  - zui-option-group
---
## What is a dropdown select?

A dropdown select is a type of input field in which a user fills an input box with an item from a list of acceptable items. Dropdown selects allow only one option to be chosen.

- - -

## Usage

Use a dropdown select to organize a list of options a user can select from.

![Dropdown select usage](/images/select_usage.svg)

- - -

## Anatomy

The anatomy of a dropdown select.

![Dropdown select anatomy](/images/select_anatomy.svg)

> 1. **Input:** The field designated for selecting an individual option from a grouping of related options.

Reviewer: ```suggestion
1. **Select:** The field designated for selecting an individual option from a grouping of related options.
```
Even though this is technically an input field to allow users to type in, we should still recognize it as a select box to not confuse it with the input field.

2. **Chevron:** The chevron arrow, within the input suggesting that there are items to select within a dropdown.
3. **Option list:** The grouping of related input options.
4. **Option:** A unique input option.

> 5. **Scrollbar**: A bar used to scroll to view more options

Reviewer: ```suggestion
5. **Scrollbar**: A bar used to scroll to view more options.
```

<br>

See the [Design specs](https://xd.adobe.com/view/ef2f902b-219f-4e41-8bba-2bf079fc5969-ba7c/grid) for detailed sizing and spacing information

- - -

## Types

#### Grouping

> * Titles can break up options into logical groups to help users scan the options more easily.

Reviewer: ```suggestion
* Grouping options with proper group labels can help users scan the options more easily.
```

![Dropdown select grouping](/images/select_grouping_specs.svg)

<br>

#### Icons

* Use icons to distinguish between options.

![Dropdown select icons](/images/select_icon_specs.svg)

- - -

## States

> #### Input states

Reviewer: ```suggestion
#### Select states
```

> Dropdown select can display the following states: unselected, hover, focused, active, selected and disabled.

Reviewer: ```suggestion
Dropdown selects can display the following states: unselected, hover, focused, active, selected and disabled.
```

![Dropdown select states](/images/select_states.svg)

#### Option list states

> Dropdown select option list can display the following states: selected, hover and disabled.

Reviewer: ```suggestion
Dropdown select option lists can display the following states: selected, hover and disabled.
```

![Dropdown select list states](/images/select_list_states.svg)

- - -

## Behavior

Sort the list in a logical order. For instance, put the most selected option(s) at the top. In many cases we recommend alphabetical order.

#### Open / close

> * Normal behavior is to reveal the list down when placement of the select input is high enough within the viewport. When select input is lower on the screen the list will reveal up.

Reviewer: ```suggestion
* Normal behavior is to reveal the option list below the select box when there is sufficient space within the viewport to display it entirely. When lower in the viewport, the option list should reveal above the select box.
```

> * Dropdown chevron icon: Rotates 180 degrees clockwise when opened, when closed spins back 180 degrees counter clockwise, returning to the original position.

Reviewer: ```suggestion
* Chevron behavior: When the option list is opened, the chevron rotates 180 degrees clockwise. When the option list is closed, the chevron spins back 180 degrees counter clockwise, returning to its original position.
```

> * The dropdown will close automatically after making a selection.

Reviewer: ```suggestion
* The option list will close automatically after making a selection or when the user clicks outside of the Dropdown select element.
```

#### Scrolling

> * If the options can be displayed within the viewport it is ideal to not have a scroll bar. Scroll bars should appear when options lists exceed the height of the screen but should not be taller than 300px.
> * Note options lists will appear below or above select inputs depending on where they sit within the viewport (see Open / close).

Reviewer: ```suggestion
* If all options can be displayed within the viewport it is ideal to not have a scroll bar. Scroll bars should appear when option lists exceed the height of the screen, but should not be taller than 300px.
* Note: option lists will appear below or above the select boxes depending on where they sit within the viewport. (See Open / close)
```

#### Typeahead

* Present in all dropdown selects, streamlines experience for users.

> * Typeahead is relevant only to options in the list, it will not return group names. Grouping headers are not seen when using typeahead.

Reviewer: ```suggestion
* Typeahead is relevant only to options in the list, it will not return group labels. Group labels are not seen when using typeahead.
```

[See typeahead](/design-system/patterns/typeahead/)

#### Defaulting a selection

> * A dropdown select shouldn't default a selected option. If you're confident users will likely choose an option you may consider defaulting to a selection.

Reviewer: ```suggestion
* A Dropdown select shouldn't default a selected option. Defaulting a selection only makes sense after a user's selection has been saved and they revisit to update their choice.
```
The last sentence isn't necessary IMO. We actually want them to not default a selection, unless it's to come back and make changes from a previous save.
